### PR TITLE
Use the LazyUploadingTaskReader #minor

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
This PR uses latest flyteplugins - pluginmachinery interface, that replaces a simple TaskReader with a TaskReader that has a path interface. The path method returns a deterministic path that can be passed to the actual container task to execute.

Depends on: https://github.com/flyteorg/flyteplugins/pull/166

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/850

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ propeller:
       type: bucket
       rate: 100
       capacity: 1000
-  kube-config: "$HOME/.kube/config"
+  kube-config: "$HOME/kubeconfig/k3s/k3s.yaml"
```
Comment (EngHabu) [inline config.yaml:25]:
```suggestion
  # This config assumes using `make start` in flytesnacks repo to startup a DinD k3s container
  kube-config: "$HOME/kubeconfig/k3s/k3s.yaml"
```

Comment (kumare3) [reply config.yaml:26]: done

**Review by EngHabu — APPROVED** [review]
